### PR TITLE
[DEV-1234] woff2 font format is not loaded

### PR DESCRIPTION
**Pull request by jeremygordillo**
#### List of Changes

Updated lambda checks to prevent `.woff2` font requests from being modified

#### Motivation and Context

This change is required to load fonts with `.woff2` format

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/cloudfront-functions/src/viewer-request-handler.ts [hunk 1]
```diff
@@ -15,9 +15,15 @@ const handler = (
...
     const isGitbookAssets = uri.startsWith('/gitbook/docs');
+    const isWoff2 = uri.endsWith('.woff2'); // woff2 is a font format
 
     // Add the .html extension if missing
-    if (!uri.endsWith('/') && !isGitbookAssets && !/\.[a-zA-Z]+$/.test(uri)) {
+    if (
+      !uri.endsWith('/') &&
+      !isGitbookAssets &&
+      !isWoff2 &&
+      !/\.[a-zA-Z]+$/.test(uri)
```
Comment (datalek) [inline apps/cloudfront-functions/src/viewer-request-handler.ts:25]: I would generalize it as follows:

```suggestion
      !/\.[a-zA-Z0-9]+$/.test(uri)
```

File: apps/cloudfront-functions/src/viewer-request-handler.ts [hunk 2]
```diff
@@ -15,9 +15,15 @@ const handler = (
 
     // Check if the uri refers to a gitbook assets
     const isGitbookAssets = uri.startsWith('/gitbook/docs');
+    const isWoff2 = uri.endsWith('.woff2'); // woff2 is a font format
```
Comment (datalek) [inline apps/cloudfront-functions/src/viewer-request-handler.ts:18]: If you change the regex, this should no longer be needed

Comment (jeremygordillo) [reply apps/cloudfront-functions/src/viewer-request-handler.ts:18]: If I change the regex, this test case breaks:

```js
const example1 = '/i-s/g/mo/v1.0';
expect(handler(makeEvent(example1)).uri).toBe(`${example1}.html`);
```

Comment (datalek) [reply apps/cloudfront-functions/src/viewer-request-handler.ts:18]: Great, then it's better to specify the extension as you've already done

File: .changeset/eighty-toys-care.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"cloudfront-functions": patch
+---
+
+handle woff2 font request
```
Comment (datalek) [inline .changeset/eighty-toys-care.md:5]:
```suggestion
Handle woff2 font request
```

**Review by datalek — APPROVED** [review]

**Review by datalek — APPROVED** [review]

**Pull request merged**